From 1f41b740419d3aaa47b6b2cf5ca737ad5f7831e6 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Wed, 20 Oct 2021 09:49:28 +0000
Subject: [PATCH backport 5.10 050/197] mlxsw: Use weak reverse dependencies
 for firmware flashing selection

Replace configuration options "select" by "imply". It allows to set
'MLXFW' symbol to 'n' from a direct dependency or with a visible
prompt.

User might wish to disable this option in case only 'mlxsw_minimal' is
configured, since it does not support firmware flashing.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/Kconfig b/drivers/net/ethernet/mellanox/mlxsw/Kconfig
index 12871c8dc7c1..a0d5f5b6c949 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/Kconfig
+++ b/drivers/net/ethernet/mellanox/mlxsw/Kconfig
@@ -6,7 +6,7 @@
 config MLXSW_CORE
 	tristate "Mellanox Technologies Switch ASICs support"
 	select NET_DEVLINK
-	select MLXFW
+	imply MLXFW
 	help
 	  This driver supports Mellanox Technologies Switch ASICs family.
 
-- 
2.20.1

